safe_eth/safe/api/base_api.py

```diff
@@ -16,52 +16,74 @@ class SafeAPIException(Exception):
 
 
 class SafeBaseAPI(ABC):
-    URL_BY_NETWORK: Dict[EthereumNetwork, str] = {}
-
     def __init__(
         self,
         network: EthereumNetwork,
         ethereum_client: Optional[EthereumClient] = None,
         base_url: Optional[str] = None,
+        api_key: Optional[str] = None,
         request_timeout: int = 10,
     ):
         """
         :param network: Network for the transaction service
         :param ethereum_client:
         :param base_url: If a custom transaction service is used
+        :param api_key: Api key of authenticated Safe services
         :raises: EthereumNetworkNotSupported
         """
         self.network = network
         self.ethereum_client = ethereum_client
-        self.base_url = base_url or self.URL_BY_NETWORK.get(network, "")
-        if not self.base_url:
+        base_url = base_url or self.get_url_by_network(network)
+        if not base_url:
             raise EthereumNetworkNotSupported(network)
+        self.base_url = base_url
```

Comment:
```suggestion
        self.base_url = base_url or self.get_url_by_network(network)
```
Also I think this exception `raise EthereumNetworkNotSupported(network)`, should be raised by `get_url_by_network`.

Reply:
Since this method must be implemented in the child class and not in this one, I have added a new method that calls the abstract method that implements the url per network and does the exception handling. https://github.com/safe-global/safe-eth-py/pull/1752/commits/d582274b25482061d5b747b1ad79d848a9b9978b